### ISSUE TITLE
Modify the implementation of permutations for static label flipping

### ISSUE THREAD

**Issue by matol-16** [EA-Robust-AI-X2023/byzfl#8]
#3 
For [static label flipping](https://github.com/EA-Robust-AI-X2023/byzfl/blob/variance_and_plot/byzfl/poisoning/attacks.py), we use a permutation specified in the config.json file:

`class StaticLabelFlipping(object): 
    """
    Description
    -----------

    Flip target labels with a fixed permutation.

    Initialization parameters
    --------------------------

    permutation: dict
        A dictionary that maps a label to its flipped version.

    Calling the instance
    --------------------

    Input parameters
    ----------------

    model : torch.nn.Module
        The local model.
    inputs : numpy.ndarray or torch.Tensor
        A collection of input features.
    targets : numpy.ndarray or torch.Tensor
        A collection of targets.

    Returns
    -------
    tuple
        The inputs and targets with flipped labels.
    """
    def __init__(self, permutation):
        if not isinstance(permutation, dict):
            raise TypeError("permutation must be a dict.")
        #convert all values in permutation to integers. Not optimal but best to avoid bugs
        permutation_int={}
        for i,j in permutation.items():
            permutation_int[int(i)]=int(j)
        self.permutation=permutation_int
    
    def __call__(self, model, inputs, targets):
        for i, target in enumerate(targets.numpy()):
            targets[i] = self.permutation[target]
        
        return inputs, targets`

But this is not very robust and scalable. In the config.json file, this implies specifying the permutation by hand: `"attack": [
    {
      "name": "StaticLabelFlipping",
      "p": 1.0,
      "parameters": {
        "permutation": {
          "0": 9,
          "1": 8,
          "2": 7,
          "3": 6,
          "4": 5,
          "5": 4,
          "6": 3,
          "7": 2,
          "8": 1,
          "9": 0
        }
      }
    }
  ],`

We could implement a default permutation (the one used in peng et al, e.g. class k flipped to nb_classes - k.

We could also implement several permutations with different names: for instance random permutation etc... This would allow to test whether the static label flipping for the permutation used in Peng et al. is an exception or not. Indeed, in Peng et all, classes 0 and 9 are flipped by the poisonned worker, which is precisely the 10th worker. But 9 and 0 are quite similar. Using other flips (0 -> 7 for instance) could change the results.
